coap-gateway/service/clientObserveHandler.go

```diff
@@ -68,30 +68,35 @@ func CreateResourceContentToObserver(client *session, resourceChanged *events.Re
 }
 
 type resourceSubscription struct {
-	client   *session
-	token    coapMessage.Token
-	authCtx  *authorizationContext
-	deviceID string
-	href     string
-
-	seqNum uint32
-	sub    *subscription.Sub
-
-	mutex              sync.Mutex
+	client             *session
+	authCtx            *authorizationContext
+	sub                *subscription.Sub
+	deviceID           string
+	href               string
+	token              coapMessage.Token
 	version            uint64
+	mutex              sync.Mutex
```

Comment:
I like the convention of putting the mutex and the members it protects in a sub structure
```
private struct {
  mutex sync.Mutex
  version uint64
  versionInitialized bool
}`
```

But that won't shrink the size, I guess we can at least add comment `guarded by mutex` to members protected by the mutex

Reply:
moved to private struct